### PR TITLE
Unit tests - Adding unit tests to sha256_op and strarr_size

### PR DESCRIPTION
**Pull request by san1526**
|Related issue|
|---|
|#11754|

## Description
Adding unit testing to the sha256_op (similar to existing testing for other checksums) and strarr_size in string_op

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

- Memory tests for Linux
  - [x] Scan-build report
  - [x] AddressSanitizer

- [x] Added unit tests (for new features)

### CODE REVIEW

**Comment by san1526** [comment]
scanbuild
sha256tests
stringtests

File: src/unit_tests/shared/test_string_op.c [hunk 1]
```diff
@@ -809,6 +823,11 @@ int main(void) {
         cmocka_unit_test(test_os_shell_avoid_escape_all),
         cmocka_unit_test(test_os_shell_escape_backslash),
         cmocka_unit_test(test_os_shell_double_escape),
+
```
Comment (pereyra-m) [inline src/unit_tests/shared/test_string_op.c:826]: We could add here the `/* Tests strarray_size */` comment

File: src/unit_tests/shared/test_string_op.c [hunk 2]
```diff
@@ -737,6 +737,20 @@ void test_os_shell_double_escape(void ** state) {
     os_free(ret2);
 }
 
+void test_strarray_size_null() {
```
Comment (pereyra-m) [inline src/unit_tests/shared/test_string_op.c:740]: We could add here the `/* Tests strarray_size */` comment

File: src/unit_tests/os_crypto/sha256/test_sha256_op.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    const char *string_sha256_chopped = "3c8727e019";
+    int chopped_size = 10;
+    os_sha256 buffer;
+
+    OS_SHA256_String_sized(string, buffer, chopped_size);
+
+    assert_string_equal(buffer, string_sha256_chopped);
+}
+
+void test_sha256_file() {
```
Comment (pereyra-m) [inline src/unit_tests/os_crypto/sha256/test_sha256_op.c:42]: Here, we need to wrap all the necessary methods to make  `OS_SHA256_File()` thinks it's accessing a real file.
We shouldn't create a temporary file.

File: src/unit_tests/os_crypto/sha256/test_sha256_op.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    write(fd, string, strlen(string));
+    close(fd);
+
+    os_sha256 buffer;
+    assert_int_equal(OS_SHA256_File(file_name, buffer, OS_TEXT), 0);
+
+    assert_string_equal(buffer, string_sha256);
+}
+
+void test_sha256_file_fail() {
```
Comment (pereyra-m) [inline src/unit_tests/os_crypto/sha256/test_sha256_op.c:60]: Same comment about wrapping as before.

**Review by pereyra-m — CHANGES_REQUESTED** [review]
GJ!

We only need to change the way some tests are proposed.

**Review by pereyra-m — APPROVED** [review]
LGTM!
Good job fixing the other tests.

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**